interfaces/builtin/desktop.go

```diff
@@ -338,6 +349,23 @@ dbus (receive, send)
 
 /etc/xdg/user-dirs.conf r,
 /etc/xdg/user-dirs.defaults r,
+
+# TODO: allow it only in CORE DESKTOP
+# Allow use of snapd's internal 'xdg-settings'
+/usr/bin/xdg-settings ixr,
+dbus (send)
+    bus=session
+    path=/io/snapcraft/Settings
+    interface=io.snapcraft.Settings
+    member={Check,CheckSub,Get,GetSub,Set,SetSub}
+    peer=(label=unconfined),
+dbus (send)
+    bus=session
+    path=/io/snapcraft/Launcher
+    interface=io.snapcraft.Launcher
+    member={OpenURL,OpenFile}
+    peer=(label=unconfined),
+# end TODO
```

Comment:
Firefox wants to set the URL handler to itself,so it needs to access this DBus interface through the xdg-settings.

Reply:
This addition is in the `desktopPermanentSlotAppArmor` variable. The Firefox snap does not have a desktop slot, so it would never see this rule. Instead it would affect code running in the ubuntu-desktop-session snap.

And I don't think there is anything that calls xdg-settings there: they're using the glib implementation of those XDG specs rather than shell scripts.

Reply:
Mmm... or maybe I just mixed a piece of another patch... let me do a test...

Reply:
Ok, confirmed: that piece isn't needed. Removing it. Thanks!

Reply:
Yep. `snap userd` is running unconfined (as is the non-GUI `xdg-desktop-portal` process).